### PR TITLE
fix(#1605): add missing icon properties to react component

### PR DESCRIPTION
**Pull request by bdfranck**
This PR addresses https://github.com/GovAlta/ui-components/issues/1605 by making the following changes:

- Adds the following properties to the icon react component:
  - inverted
  - fillcolor
  - opacity
  - arialabel
  - title
- Adds a console warning that `ariacontrols` and `ariaexpand` are deprecated

### PR Checklist
- [x] I have run a build locally.
- [x] I have ran unit tests locally.
- [x] I have tested the functionality.
image

### CODE REVIEW

**Comment by bdfranck** [comment]
I've tested the react component...

### ✅  I can add the `inverted` property

`<GoAIcon type="add" inverted="true" />`

image

image

### ✅  I can add the `fillcolor` property

`<GoAIcon type="airplane" theme="filled" fillcolor="red" />`

image

image

### ✅  I can add the `opacity` property

`<GoAIcon type="accessibility" opacity={0.5} />`

image

image

### ✅  I can add the `arialabel` property

`<GoAIcon type="analytics" arialabel="This ia an aria label" />`

image

image

### ✅  I can add the `title` property

`<GoAIcon type="albums" title="Icon title" />`

image

image

**Comment by bdfranck** [comment]
I've tested the console warnings...

### ✅ When I use the `ariacontrols` property, I see a warning.

`<goa-icon type="accessibility" ariacontrols="foo"></goa-icon>`

image

### ✅ When I use the `ariaexpand ` property, I see a warning.

`<goa-icon type="accessibility" ariaexpanded="foo"></goa-icon>`

image

**Comment by lizhuomeng71** [comment]
Issue is verified for react and angular, for chrome, firefox and edge

image

File: libs/react-components/src/lib/icon/icon.tsx [hunk 1]
```diff
@@ -549,19 +549,35 @@ export interface GoAIconProps extends Margins {
   type: GoAIconType;
   size?: GoAIconSize;
   theme?: GoAIconTheme;
+  inverted?: string;
+  fillcolor?: string;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/icon/icon.tsx:553]: The React props use the camelCase notation, so `fillColor` and `ariaLabel`

**Review by ArakTaiRoth — APPROVED** [review]

File: libs/react-components/src/lib/icon/icon.tsx [hunk 2]
```diff
@@ -549,19 +549,35 @@ export interface GoAIconProps extends Margins {
...
 interface WCProps extends Margins {
   type: GoAIconType;
   theme?: GoAIconTheme;
   size?: GoAIconSize;
+  inverted?: string;
+  fillColor?: string;
+  opacity?: number;
+  title?: string;
+  ariaLabel?: string;
+  testId?: string;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/icon/icon.tsx:569]: @bdfranck Sorry for the confusion, the React properties should be camel case, but the web-components **need** to be camel case.

Comment (bdfranck) [reply libs/react-components/src/lib/icon/icon.tsx]: Makes sense! I've updated my PR so the `GoAIconProps` are camel case but the `WCProps` are lower case.

**Review by chrisolsen — APPROVED** [review]

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**